lib/commands/update.ts

```diff
@@ -66,42 +51,79 @@ export class UpdateCommand implements ICommand {
 		return args.length < 2 && this.$projectData.projectDir !== "";
 	}
 
-	private async executeCore(args: string[], folders: string[]): Promise<void> {
-		let platforms = this.$platformService.getInstalledPlatforms(this.$projectData);
-		const availablePlatforms = this.$platformService.getAvailablePlatforms(this.$projectData);
-		const packagePlatforms: string[] = [];
+	private async executeCore(args: string[]): Promise<void> {
+		const platforms = this.getPlatforms();
 
-		for (const platform of availablePlatforms) {
+		for (const platform of _.xor(platforms.installed, platforms.packagePlatforms)) {
```

Comment:
The logic has changed a bit here - instead of iterating over all `availablePlatforms` we are basically iterating over `packagePlatforms` because `platforms.packagePlatforms` is `platforms.installed` concatenated with `packagePlatforms` and the `xor` will basically only return `packagePlatforms`.
Is this expected

Reply:
Yes this is intentional, because we are missing the following part:
```typescript
if (platformVersion) {
 	packagePlatforms.push(platform);
 	this.$projectDataService.removeNSProperty(this.$projectData.projectDir, platformData.frameworkPackageName);
}
```

The if statement is moved to `getPlatforms` and we want to call `removeNSProperty` only for the ones that passed the condition.

The behavior should be the same.